test/groovy/util/JenkinsReadYamlRule.groovy

```diff
@@ -25,11 +25,14 @@ class JenkinsReadYamlRule implements TestRule {
         return new Statement() {
             @Override
             void evaluate() throws Throwable {
-                testInstance.helper.registerAllowedMethod("readYaml", [Map], { Map m ->
-                    if(m.text) {
-                        return new Yaml().load(m.text)
-                    } else if(m.file) {
-                        throw new UnsupportedOperationException()
+                testInstance.helper.registerAllowedMethod("readYaml", [Map], { Map parameters ->
+                    if(parameters.text) {
+                        return new Yaml().load(parameters.text)
+                    } else if(parameters.file) {
+                        if (parameters.file == '.pipeline/config.yml') {
+                            return new Yaml().load(new File('test/resources/test_pipeline_config.yml').text)
+                        }
+                        return new Yaml().load(new File(parameters.file).text)
                     } else {
                         throw new IllegalArgumentException("Key 'text' is missing in map ${m}.")
```

Comment:
The message here needs to be adjusted in order to take the new file parameter into account.